static/src/javascripts/projects/common/modules/identity/upsell/consent-card/FollowCard.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+import React, { Component } from 'preact-compat';
+import { FollowButtonWrap } from 'common/modules/identity/follow/FollowButtonWrap';
+
+/**
+ * Type for things that can be rendered in a Follow Card
+ */
+export type CardLike = {
+    id: string,
+    name: string,
+    description: string,
+};
+
+export type Consent = {
+    id: string,
+    name: string,
+    description: string,
+    isOptOut: boolean,
+    isChannel: boolean,
+};
+
+export type Followable<T: CardLike> = {
+    value: T,
+    onChange: boolean => void,
+};
+
+type FollowCardProps<T: CardLike> = {
```

Comment:
Maybe one of the flow pros from @guardian/guardian-frontend-team could see if this is a reasonable thing to do in JavaScript land?